execute/dispatcher.go

```diff
@@ -43,17 +45,24 @@ type poolDispatcher struct {
 func newPoolDispatcher(throughput int, logger *zap.Logger) *poolDispatcher {
 	return &poolDispatcher{
 		throughput: throughput,
-		work:       make(chan ScheduleFunc, 100),
+		work:       newRing(100),
```

Comment:
looks like we are still defaulting the buffer to 100 like it was in the existing code. But when it goes above 100 then we manually increase the buffer size and manually copy the buffer.

Reply:
Yes. It's a resizable ring buffer.